src/einsteinpy/geodesic/geodesic.py

```diff
@@ -99,13 +99,18 @@ def __init__(
         }
 
         if metric not in _METRICS:
-            raise NotImplementedError(
-                f"'{metric}' is unsupported. Currently, these metrics are supported:\
-                \n1. Schwarzschild\n2. Kerr\n3. KerrNewman"
-            )
+            if not callable(metric):
+                raise NotImplementedError(
+                    f"'{metric}' is unsupported. Currently, these metrics are supported:\
+                    \n1. Schwarzschild\n2. Kerr\n3. KerrNewman\n4. Any user-supplied callable metric function"
+                )
+
+            self.metric_name = metric.__name__
```

Comment:
L108 & L109 are currently not being covered by tests, hence the Codecov error. Add a simple test with a custom metric function around this [unit-test](https://github.com/a3ahmad/einsteinpy/blob/b1934ef6c1cbb3b7a18a6837cdd514a2646ef4c8/tests/test_geodesic/test_geodesic.py#L53). Feel free to ask, if you have any questions.

Reply:
Fixed now, sorry about that. I had made a mental note to include a unit-test, but forgot.

I updated the changelog, too. I assumed I needed to put 625 as the ref #, given that that's the PR number?

Reply:
> I updated the changelog, too. I assumed I needed to put 625 as the ref #, given that that's the PR number?

That's right.